cpp/src/arrow/testing/gtest_util.h

```diff
@@ -24,6 +24,7 @@
 #include <functional>
 #include <memory>
 #include <string>
+#include <string_view>
```

Comment:
```suggestion
#include <string_view>
#include <optional>
```

Reply:
We'll want to delete the `nonstd` string_view and optional impls to be sure we don't have both lying around. Similarly, we can remove `util::Variant` in favor of `std::variant`